src/components/QueryEditor.tsx

```diff
@@ -436,16 +437,37 @@ const getConditionByUid = (
 type QueryEditorComponent = (props: {
   parentUid: string;
   onQuery?: () => void;
+  setHasResults?: () => void;
   hideCustomSwitch?: boolean;
   showAlias?: boolean;
 }) => JSX.Element;
 
 const QueryEditor: QueryEditorComponent = ({
   parentUid,
   onQuery,
+  setHasResults,
   hideCustomSwitch,
   showAlias,
 }) => {
+  const previewQuery = useCallback(
+    (e) => {
+      if (e.altKey && e.key === "q") {
+        const target = document.activeElement as HTMLElement;
+        const uid = getBlockUidFromTarget(target);
+        if (uid !== parentUid) return;
+        if (setHasResults) setHasResults();
+      }
```

Comment:
- `commandPalette.addCommand()` in `index.ts` so we only set it once (can keep the keyboard shortcut as a default).
- On trigger, `const target = document.activeElement as HTMLElement;const uid = getBlockUidFromTarget(target);`
- end it by calling `document.body.dispatch(new CustomEvent("roamjs-query-builder:fire-query", {detail: uid}))`
- the use effect below is replaces `window.addEventListener` with `document.body.addEventListener("roamjs-query-builder:fire-query")` - `if (uid !== parentUid) return;`